File: core/ShuffleAnArray/ShuffleAnArray.java
package com.dbc.code;

import java.util.Random;

public class ShuffleAnArray {
    private int[] nums = null;
    private int[] shuffleNums = null;
    private final Random random = new Random();

    public ShuffleAnArray(int[] nums) {
        this.nums = nums;
        this.shuffleNums = this.nums.clone();
    }

    public int[] reset() {
        this.shuffleNums = this.nums.clone();
        return this.shuffleNums;
    }

    public int[] shuffle() {
        for (int i = 0; i < this.nums.length; i++) {
            int point = random.nextInt(this.nums.length - i) + i;
            int temp = this.shuffleNums[point];
            this.shuffleNums[point] = this.shuffleNums[i];
            this.shuffleNums[i] = temp;
        }
        return this.shuffleNums;
    }
}

File: core/ShuffleAnArray/ShuffleAnArray.py
import random
from typing import List
from copy import deepcopy
from random import randint

class Solution:

    def __init__(self, nums: List[int]):
        self.nums = nums
        self.shuffle_nums = deepcopy(self.nums)

    def reset(self) -> List[int]:
        self.shuffle_nums = deepcopy(self.nums)
        return self.shuffle_nums

    def shuffle(self) -> List[int]:
        for i in range(len(self.nums) - 1):
            point = randint(i, len(self.nums) - 1)
            self.shuffle_nums[point], self.shuffle_nums[i] = self.shuffle_nums[i], self.shuffle_nums[point]

        return self.shuffle_nums




